### PR TITLE
create jmh jar

### PR DESCRIPTION
**Pull request by adarsh-jaiswal**
In order ro run benchmark tests by the test runner we need the jmh jar. To create the jmh jar I have added the jmh plugin in this PR. Also, added exception to be thrown by the benchmark methods.

### CODE REVIEW

File: src/test/java/benchmark/AddError.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public class AddError {
     @BenchmarkMode(Mode.SingleShotTime)
     @Warmup(iterations = 1, batchSize = 50000)
     @Measurement(iterations = 1, batchSize = 5000)
-    public ExecutionContext benchMarkAddError() {
+    public ExecutionContext benchMarkAddError() throws InterruptedException {
```
Comment (chriswr95) [inline src/test/java/benchmark/AddError.java:31]: why do we need to add these exceptions to every method signature?

**Comment by chriswr95** [comment]
what is meaningfully different after this PR? Can I now run some command that I couldn't previously?

File: src/test/java/benchmark/AddError.java (hunk 1 above)
Comment (adarsh-jaiswal) [reply src/test/java/benchmark/AddError.java:31]: The generated jmh test class uses these methods and it is wrapped inside a try statement which catches InterruptedException. When I build the jmh test jar it throws compilation error. In order to fix it I had to add it to all the methods.

**Comment by adarsh-jaiswal** [comment]
> what is meaningfully different after this PR? Can I now run some command that I couldn't previously?

we can now run the bollowing commands -

1.  to build jmh jar - `RELEASE_VERSION=test-runner-jmh ./gradlew jmhJar`
2.  to run test and produce test result in json - `java -jar build/libs/graphql-java-test-runner-jmh.jar -rf json`

**Comment by andimarek** [comment]
I needed to a bit more config last time I did this: https://github.com/graphql-java/graphql-java/pull/2364/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R86
and
https://github.com/graphql-java/graphql-java/pull/2364/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R109

We don't need this?

File: src/test/java/benchmark/AddError.java (hunk 1 above)
Comment (andimarek) [reply src/test/java/benchmark/AddError.java:31]: I didn't need to do that last time. Maybe you need the additional gradle changes I mentioned here: https://github.com/graphql-java/graphql-java/pull/2871#issuecomment-1164909737

**Comment by andimarek** [comment]
@adarsh-jaiswal I was looking into this a bit more and it seemed that the jmh plugin is a bit outdated. We need to update the jmh version the plugin is using with:

```
dependencies {
    testImplementation 'org.openjdk.jmh:jmh-core:1.35'
    testAnnotationProcessor 'org.openjdk.jmh:jmh-generator-annprocess:1.35'
    jmh 'org.openjdk.jmh:jmh-core:1.35'
    jmh 'org.openjdk.jmh:jmh-generator-annprocess:1.35'
}
```
The first two dependencies are already there, but I upgraded to 1.35 jmh.
The last two are special instructions for the jmh plugin

After this it worked for me without adding any `throws InterrruptedException`.

**Pull request merged**